### ISSUE TITLE
Segmentation fault before entering tests with 0.8-rc1-2712

### ISSUE THREAD

**Issue by soywiz** [JetBrains/kotlin-native#1770]
https://github.com/korlibs/korlibs/tree/segmentation.fault.before.tests.korag

![screen shot 2018-07-06 at 04 15 37](https://user-images.githubusercontent.com/570848/42356541-4c750c44-80d3-11e8-8843-fc17415b48ba.png)

```
class ImageFormats(formats: Iterable<ImageFormat>) : ImageFormat("") {
	val formats = formats.toSet()
        // ...
}
object JPEG : ImageFormat
object PNG : ImageFormat
object TGA : ImageFormat
object PSD : ImageFormat
val StandardImageFormats = listOf(JPEG, PNG, TGA, PSD)
val defaultImageFormats = ImageFormats(StandardImageFormats.toSet())
```

**Comment by olonho** [comment]
Cannot reproduce with 
```
import kotlin.test.*
open class ImageFormat(val name: String)

class ImageFormats(formats: Iterable<ImageFormat>) : ImageFormat("") {
  val formats = formats.toSet()
}
object JPEG : ImageFormat("JPEG")
object PNG : ImageFormat("PNG")
object TGA : ImageFormat("TGA")
object PSD : ImageFormat("PSD")
val StandardImageFormats = listOf(JPEG, PNG, TGA, PSD)
val defaultImageFormats = ImageFormats(StandardImageFormats.toSet())

@Test
fun test() {
  println("OK")
}
```
Could you please provide failing test case?

**Comment by olonho** [comment]
From crash it looks like you somehow managed to get incorrect init order, when defaultImageFormats is inited before StandardImageFormats.

**Comment by soywiz** [comment]
Yeah, since relevant code was in several files, I just provided a set of sample statements to see an overview of what I wrote. But probably the problem is other.

The way to reproduce it is to download the branch I linked, and run `./gradlew check`.
The executable with the problem generated was not even direct tests for the library, but test for another library depending on that one:

```
./gradlew check
./korag/korag/native/build/exe/test/debug/test.kexe # should be this one as far as I remember
```

In master I changed it completely so I do not have this code anymore. But still would be nice to know why the initialization order was wrong in that case, that's why I created that branch to be able to reproduce it.
